lib/msgpack/timestamp.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module MessagePack
+  Timestamp = Struct.new(:sec, :nsec)
+
+  class Timestamp # a.k.a. "TimeSpec"
```

Comment:
I can't find how to use the `Struct.new BLOCK` syntax with constants:

NG(1):

```
$ ruby -e 'C = Struct.new(:foo) do CONST = 42 end; pp(C::CONST)'
Traceback (most recent call last):
-e:1:in `<main>': uninitialized constant C::CONST (NameError)
```

NG(2):

```
$ ruby -e 'Struct.new("::C", :foo) do pp self; CONST = 42 end; pp(C::CONST)'
Traceback (most recent call last):
	1: from -e:1:in `<main>'
-e:1:in `new': identifier ::C needs to be constant (NameError)
```

NG(3; it's not what I want):

```
$ ruby -e 'Struct.new("C", :foo) do pp self; self::CONST = 42 end; pp(Struct::C::CONST)'
Struct::C
42
```
I'm sorry I'm not good at Ruby 😭

Reply:
Do you mean that there is no way to refer `MessagePack::Timestamp::TYPE` from other classes?
It's not impossible (for example, `C = Struct.new(); C.const_set(:CONST, 42); p C::CONST`), but I guess we should define normal modules/classes instead of using Struct.
Is there any reason to use Struct?

Reply:
I use Struct because of its auto-generated `#==` method, which is useful only for testing.

But yes, I'll use a normal class; it's just easy.

Reply:
done: 733ed14